plaso/parsers/text_plugins/cri.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""Text file parser plugin for Container Runtime Interface (CRI) log format.
+
+This is a text-based log format used in kubernetes/GKE.
+
+Also see:
+  https://github.com/kubernetes/design-proposals-archive/blob/main/node/kubelet-cri-logging.md
+"""
+
+import pyparsing
+
+from dfdatetime import golang_time
+
+from plaso.containers import events
+from plaso.lib import errors
+from plaso.parsers import text_parser
+from plaso.parsers.text_plugins import interface
+
+
+class CRIEventData(events.EventData):
+  """CRI log event data.
+
+  Attributes:
+    event_datetime (dfdatetime.Golangtime): the datetime of the log message.
+    message (str): the log message.
+    stream (str): the log stream.  Currently only 'stdout' and 'stderr' are
+        supported.
+    tag (str): the log tag.  Currently only 'P' (partial) and 'F' (full) are
+        supported.
+  """
+  DATA_TYPE = 'cri:container:log:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(CRIEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.event_datetime = None
+    self.message = None
+    self.stream = None
+    self.tag = None
+
+
+class CRITextPlugin(interface.TextPlugin):
+  """Text file parser plugin for CRI log files."""
+
+  NAME = 'cri_log'
+  DATA_FORMAT = 'Container Runtime Interface log file'
+
+  ENCODING = 'utf-8'
+
+  _TWO_DIGITS = pyparsing.Word(pyparsing.nums, exact=2)
+
+  _FOUR_DIGITS = pyparsing.Word(pyparsing.nums, exact=4)
+
+  _NINE_DIGITS = pyparsing.Word(pyparsing.nums, exact=9)
+
+  # Date and time values are formatted as: 2016-10-06T00:17:09.669794202Z
+  _DATE_AND_TIME = pyparsing.Combine(
```

Comment:
I recommend to make this a single regex, which appears to be more performant.

Reply:
Done